### PR TITLE
Feature/room ownership

### PR DESCRIPTION
**Pull request by brianpeiris**
- Adds a sign-in/out link to the home page
- Adds auth tokens to room creation and room join

### CODE REVIEW

File: src/hub.js [hunk 1]
```diff
@@ -531,6 +531,10 @@ document.addEventListener("DOMContentLoaded", async () => {
 
   const pushSubscriptionEndpoint = await subscriptions.getCurrentEndpoint();
   const joinPayload = { profile: store.state.profile, push_subscription_endpoint: pushSubscriptionEndpoint, context };
+  const { token } = store.state.credentials;
+  if (token) {
+    joinPayload.token = token;
```
Comment (gfodor) [inline src/hub.js]: notes from other PR, but these should prob be `auth_token` vs `perm_token`

File: src/react-components/home-root.js [hunk 2]
```diff
@@ -41,12 +45,18 @@ class HomeRoot extends Component {
   state = {
     environments: [],
     dialog: null,
+    signedIn: null,
```
Comment (gfodor) [inline src/react-components/home-root.js]: nit, maybe make this `authenticated` or `signedIn` everywhere, not both

File: src/react-components/scene-ui.js [hunk 3]
```diff
@@ -52,23 +51,8 @@ class SceneUI extends Component {
...
-
-    const hub = await res.json();
-
-    if (!process.env.RETICULUM_SERVER || document.location.host === process.env.RETICULUM_SERVER) {
-      document.location = hub.url;
-    } else {
-      document.location = `/hub.html?hub_id=${hub.hub_id}`;
-    }
+  createRoom = () => {
+    createAndRedirectToNewHub(this.state.customRoomName, this.props.sceneId);
```
Comment (gfodor) [inline src/react-components/scene-ui.js]: oops

**Comment by gfodor** [comment]
LGTM mod the changes to move away from `is_owner` and towards using a `perm_token` (doesn't have to block but seems like we should just do it now.)

Also, nice to see the auth channel stuff migrated over to the home page nicely. 😅

**Comment by brianpeiris** [comment]
Merging in #843

**Pull request closed without merging**